File: frontend/src/api/importTemplate.ts
// 导入规则模板管理（驱动 Obsidian 目录一键导入的闪卡/题库抽取）请求层。
import { apiGet, apiPost, apiPut, apiDelete } from './request'

/** 模板类型 */
export type TemplateType = 'FLASHCARD' | 'QUIZ' | 'PATH'

/** 模板内容结构（JSON），由编辑器生成 */
export interface ImportTemplateContent {
  /** 字段结构定义 */
  fieldSchema: Array<{
    key: string
    label: string
    type: 'text' | 'markdown' | 'json' | 'number'
    required: boolean
    source: string
  }>
  /** 抽取规则 */
  rules: {
    headingLevel?: number
    maxPerDoc?: number
    questionTypes?: string[]
  }
  /** 校验规则 */
  validation: Array<{ field: string; rule: string; value?: number }>
  /** 展示样式 */
  style: {
    cardLayout?: 'qa' | 'flip' | 'list'
    showImage?: boolean
    theme?: 'light' | 'dark'
  }
  /** 数据源绑定 */
  sourceBinding: {
    mode: 'heading' | 'tag' | 'keyword'
    pattern: string
  }
}

/** 模板视图对象（列表/详情返回） */
export interface ImportTemplateVO {
  id: number
  userId: number
  name: string
  type: TemplateType
  description?: string
  content: string
  enabled: number
  isDefault: number
  isPreset: number
  createTime?: string
  updateTime?: string
  fieldCount?: number
  validationCount?: number
  ruleSummary?: string
}

/** 创建/更新请求体 */
export interface ImportTemplatePayload {
  name: string
  type: TemplateType
  description?: string
  content: string
  enabled?: number
}

export const importTemplateApi = {
  /** 模板列表（可按类型/启用状态过滤） */
  list: (params?: { type?: TemplateType; enabled?: number }) =>
    apiGet<ImportTemplateVO[]>('/import-templates', { params }),

  /** 模板详情 */
  detail: (id: number) => apiGet<ImportTemplateVO>(`/import-templates/${id}`),

  /** 创建模板（归属当前用户） */
  create: (payload: ImportTemplatePayload) =>
    apiPost<ImportTemplateVO>('/import-templates', payload),

  /** 更新模板（仅本人非预设模板） */
  update: (id: number, payload: ImportTemplatePayload) =>
    apiPut<ImportTemplateVO>(`/import-templates/${id}`, payload),

  /** 删除模板（预设不可删） */
  remove: (id: number) => apiDelete(`/import-templates/${id}`),

  /** 启用/停用切换 */
  toggle: (id: number) => apiPost(`/import-templates/${id}/toggle`),

  /** 设为默认模板（同类型唯一） */
  setDefault: (id: number) => apiPost(`/import-templates/${id}/default`),
}

File: backend/src/main/java/com/knowflow/entity/ImportTemplate.java
package com.knowflow.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 导入规则模板实体：用于驱动 Obsidian 目录一键导入时的闪卡/题库抽取逻辑。
 * <p>
 * 模板内容以 JSON 形式存储于 {@link #content}，结构由前端规则模板编辑器定义，包含：
 * 字段结构(fieldSchema)、抽取规则(rules)、校验规则(validation)、展示样式(style)、数据源绑定(sourceBinding)。
 * 预设模板(isPreset=1)对所有用户可见；用户自定义模板(isPreset=0)归属创建者。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("import_template")
public class ImportTemplate extends BaseEntity {

    /** 创建者用户 ID（sys_user.id），预设模板为 1（系统） */
    private Long userId;

    /** 模板名称 */
    private String name;

    /** 模板类型：FLASHCARD 闪卡 / QUIZ 题库 / PATH 学习路径 */
    private String type;

    /** 模板描述 */
    private String description;

    /** 模板内容（JSON 字符串）：字段结构/规则/校验/样式/数据源绑定 */
    private String content;

    /** 是否启用：1 启用 / 0 停用 */
    private Integer enabled;

    /** 是否默认模板：1 默认 / 0 否（同类型仅一个默认） */
    private Integer isDefault;

    /** 是否预设模板：1 系统预设 / 0 用户自定义 */
    private Integer isPreset;
}
